rodeo/src/rodeo/06-either-and-try.scala

```diff
@@ -0,0 +1,186 @@
+package rodeo
+import zio.test._
+
+object EitherAndTry extends Chapter {
+
+  import scala.util.{Try, Success, Failure}
+
+  // The `Try[A]` type is similar to `Option[A]`. It is used to represent the type of a computation that attempts to return an instance of type A.
+  // If the computation succeeds we obtain an instance of type `Success[A]`. If it fails we obtain a wrapped instance of the Throwable class. This instance will have type `Failure`.
+  // The Throwable class is the least derived class of the Scala exception hierarchy. An instance of this class can represent an error or an exception.
+  // For practical purposes the main difference between `None` and `Failure` is that `Failure` is able to provide insight into the cause of the "failure".
+
+  Exercise("Try constructor") {
+    assertTrue(Try(7) == Success(???))
+
+    // Delete the following line if you can't find the answer.
+
+    assertTrue(Try(10 / 0).getClass == ???)
+  }
+
+  Exercise("Books") {
+    def bookRecomendation(
+        likesMath: Boolean,
+        likesComputerScience: Boolean
+    ): Try[String] =
+      if (!likesMath && !likesComputerScience)
+        Failure(
+          new IllegalArgumentException("You need to go visit a quantum doctor")
+        )
+      else if (likesMath && !likesComputerScience)
+        Failure(new IllegalArgumentException("You should not be reading this"))
+      else if (!likesMath && likesComputerScience)
+        Failure(
+          new IllegalArgumentException(
+            "I am unable to give you a recommendation"
+          )
+        )
+      else Success("The emperors new mind")
+
+    assertTrue(bookRecomendation(false, true) == ???)
+    assertTrue(bookRecomendation(true, true) == ???)
+  }
+
+// To find a solution of the quadratic equation ax^2+bx+c=0 we only need to use the formula  x = (-b + sqrt(b^2-4ac))/2a.
```

Comment:
move this comment right where we define the `def discriminant` and perhaps link to the wikipedia page to show some context if needed:

https://en.wikipedia.org/wiki/Discriminant#Degree_2